exportlegends.lua

```diff
@@ -47,6 +47,33 @@ local MAPS = {
     "Diplomacy",
 }
 
+-- check if a folder with this name could be created or already exists
+function get_world_date_str()
+    local month = dfhack.world.ReadCurrentMonth() + 1 --days and months are 1-indexed
+    local day = dfhack.world.ReadCurrentDay()
+    local year_str = string.format('%0'..math.max(5, string.len(''..df.global.cur_year))..'d', df.global.cur_year)
+    local date_str = year_str..string.format('-%02d-%02d', month, day)
+    return date_str
+end
+
+-- set current directory as main folder
+local main_folder = dfhack.filesystem.getcwd()
```

Comment:
I actually ran into a 9-year-old DFHack core bug when I was testing this - `dfhack.getDFPath()` is supposed to return the path to the DF root, but on Linux it was essentially returning `getcwd()` instead. With the [fix](https://github.com/DFHack/dfhack/commit/56e43a0dde023c5a4595a22b29d800153b31e3c4), I would use getDFPath instead, since that's really what you care about.

Reply:
Yeah already had the code in there but did not seem to work, but did not know it was an issue.